aconfig.go

```diff
@@ -224,19 +242,12 @@ func (l *Loader) loadFromFile(dst interface{}) error {
 		defer func() { _ = f.Close() }()
 
 		ext := strings.ToLower(filepath.Ext(file))
-		switch ext {
-		case ".yaml", ".yml":
-			err = yaml.NewDecoder(f).Decode(dst)
-		case ".json":
-			err = json.NewDecoder(f).Decode(dst)
-		case ".toml":
-			_, err = toml.DecodeReader(f, dst)
-		case ".hcl":
-			err = hclsimple.DecodeFile(file, nil, dst)
-		default:
+		d, ok := l.config.FileDecoders[ext]
+		if !ok {
 			return fmt.Errorf("file format '%q' isn't supported", ext)
 		}
 
+		err = d.DecodeFile(file, dst)
 		if err == nil {
```

Comment:
This func will stop after first succeeded `DecodeFile()`, is this intended?

Reply:
sure, you've decoded 1 file with a config, job is done, free market